controllers/storageclaim_controller.go

```diff
@@ -267,12 +266,15 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 
 		updateStorageClaim := false
 		// Check if finalizers are present, if not, add them.
-		if !contains(r.storageClaim.GetFinalizers(), storageClaimFinalizer) {
-			r.log.Info("Finalizer not found for StorageClaim. Adding finalizer.", "StorageClaim", r.storageClaim.Name)
-			r.storageClaim.SetFinalizers(append(r.storageClaim.GetFinalizers(), storageClaimFinalizer))
+		if controllerutil.AddFinalizer(r.storageClaim, storageClaimFinalizer) {
 			updateStorageClaim = true
 		}
-		if utils.AddAnnotation(r.storageClaim, storageClientAnnotationKey, client.ObjectKeyFromObject(r.storageClient).String()) {
+
+		// if claim is created by default we'll already have a controller reference, however set the owner if it is explictly created by user
+		if !controllerutil.HasControllerReference(r.storageClaim) {
+			if err := r.ownWithOwner(r.storageClient, r.storageClaim); err != nil {
```

Comment:
Claims created by the user are owned by the user, not by the client. We should not add an owner reference in that case

Reply:
Ack, so during deletion of client I'll check for all claims referring to that client then.

Reply:
Yes

Reply:
Ack, if client is not owning claim then I need to add anno/label on claim to track it.

However, if I can make use of CRD cel rule which already exist I can make use of every client having client name in its spec which also removes the other pointer to string business.

Reply:
Why? the claim already has it in the spec. Why do we need to add an additional ref?

Reply:
as stated in https://github.com/red-hat-storage/ocs-client-operator/pull/133#discussion_r1555153525, client field in claim spec can be empty.

Reply:
I see another issue, if we create claim we should be the one to delete it, if user creates it then user should delete it, however having two entities deleting similar resources may not be a good idea.

can we provide the whole deletion to user and not reconcile even if claim corresponding to a client is deleted by tracking via anno.

Reply:
done, operator removes auto created claims during client offboard.